### ISSUE TITLE
Feature Request: Support for VICE Testbench/Debugcart?

### ISSUE THREAD

**Issue by ArKay74** [MiSTer-devel/C64_MiSTer#143]
I think it would help to test the compatibility if the VICE Testbench would be supported.

https://vice-emu.pokefinder.org/wiki/Testbench

But I have no idea if it would be possible testing an FPGA that way.

**Comment by sorgelig** [comment]
as i see, almost all testbenches there are in prg format which you can run in core.

**Comment by ArKay74** [comment]
Yes, but you would have to do those >1200 tests by hand which is why there is a script that does that for you.

**Comment by ArKay74** [comment]
I'd be interested to check if there's a way. I'm a software developer, but I know nothing about how the cores are un on the MiSTer, what options there are when launching cores, etc. Would have to check and see. Currently a bit busy with other things, but I could do that if it makes sense.

**Comment by ArKay74** [comment]
There are ways of launching cores with a given program or cartridge, and there's even a way to take a screenshot (there's also a project that offers a REST API for doing those things):

https://github.com/wizzomafizzo/mrext/blob/main/docs/remote-api.md

But I don't know how to access the core's memory. I guess the core is using the memory that is on the FPGA, not the one that the linux is running on. So there's probably no way of accessing that? There's no save state support in the 64 core either.

**Comment by mrdudz** [comment]
Being able to access the RAM is not really required - what is needed is that the value written to $d7ff can be "seen" - somehow. It can be a IO register that does nothing besides providing the written value to whatever remote API (this is what U64 does)

**Comment by ArKay74** [comment]
Yeah, but that's not something I could access from the outside. That's most likely functionality that needs to be added in the core?
